tests/sonic_xcvr/test_xcvr_field.py

```diff
@@ -0,0 +1,208 @@
+        assert field.get_offset() == parent_field.get_offset()
+
+    def test_size(self):
+        field = mem_map.get_field("BitField0")
+        assert field.get_size() == 1
+
+    def test_read_before_write(self):
+        field = mem_map.get_field("BitField0")
+        assert field.read_before_write
+
+    def test_encode_decode(self):
+        field = mem_map.get_field("BitField1")
+        encoded_data = field.encode(True, bytearray(b'\x01'))
+        assert encoded_data == bytearray(b'\x03')
+
+        encoded_data = field.encode(False, bytearray(b'\x03'))
+        assert encoded_data == bytearray(b'\x01')
+
+        decoded_data = field.decode(bytearray(b'\x02'))
+        assert decoded_data
+
+        decoded_data = field.decode(bytearray(b'\x00'))
+        assert not decoded_data
+
+        assert field.decode(field.encode(True, bytearray(b'\x00')))
+        assert not field.decode(field.encode(False, bytearray(b'\x00')))
+
+class TestRegField(object):
+    def test_offset(self):
+        field = mem_map.get_field("Field0")
+        assert field.offset == 6
+
+    def test_size(self):
+        field = mem_map.get_field("Field1")
+        assert field.get_size() == 1
+
+    def test_read_before_write(self):
+        field = mem_map.get_field("Field0")
+        assert not field.read_before_write()
+
+class TestNumberRegField(object):
+    def test_encode_decode(self):
+        field = mem_map.get_field("NumReg")
+        val = 0xFFFFFFFFFFFFFFFF
+        assert field.decode(field.encode(val)) == val
+
+        field = mem_map.get_field("ScaleNumReg")
+        val = -100
+        assert field.decode(field.encode(val)) == val
+
+        field = mem_map.get_field("Field1")
+        val = 0xFF
+        assert field.decode(field.encode(val)) == val
+        assert field.decode(field.encode(0)) == 0
+
+class TestStringRegField(object):
+    def test_decode(self):
+        field = mem_map.get_field("StringReg")
+        data = bytearray("Arista Networks".encode("ascii"))
+        assert field.decode(data) == "Arista Networks"
+
+class TestCodeRegField(object):
+    def test_decode(self):
+        field = mem_map.get_field("CodeReg")
+        data = bytearray([0])
+        assert field.decode(data) == "Code0"
+
+        data = bytearray([1])
+        assert field.decode(data) == "Code1"
+
+        field = mem_map.get_field("LargeCodeReg")
+        data = bytearray([0xFE, 0x7F])
+        assert field.decode(data) == "Code0"
+        data = bytearray([0xFE, 0xFF])
+        assert field.decode(data) == "Code128"
+        data = bytearray([0xFF, 0x7F])
+        assert field.decode(data) == "Code256"
+        data = bytearray([0xFF, 0xFF])
+        assert field.decode(data) == "Code384"
+
+class TestHexRegField(object):
+    def test_decode(self):
+        field = mem_map.get_field("HexReg")
+        data = bytearray([0xAA, 0xBB, 0xCC])
+        assert field.decode(data) == "aa-bb-cc"
+
+class TestRegGroupField(object):
```

Comment:
can you add a test case to show that RegGroupField's subfields need not be contiguous and one can make use of bulk read. You can add this:-

class MockXcvrMemMap(XcvrMemMap):
    page0 = RegGroupField("Page0",
                NumberRegField("Field0", 0, format="<I", size=4),
                NumberRegField("Field31", 124, format="<I", size=4)
    

if __name__ == '__main__':
    mem_map = MockXcvrMemMap(None)

    field = mem_map.get_field("Page0")
    data = bytearray(b'\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04\x01\x02\x03\x04')
    decoded = field.decode(data)
    print(decoded)
    assert decoded == {
        "Field0" : 0x04030201,
        "Fieldn" : 0x04030201
    }

Reply:
Done